README.md
# MLM-System

Project100/history.txt
.Sat Apr  1 19:37:40 2023
  ID:H01 NAME:TAN CHIN YONG login

.Sat Apr  1 22:38:49 2023
  ID:H01 NAME:TAN CHIN YONG login
  Delete Account--> ID:S04 NAME:TABU

.Sat Apr  1 22:42:45 2023
  ID:H01 NAME:TAN CHIN YONG login
  Delete Account--> ID:S03 NAME:SITI

.Sat Apr  1 22:46:01 2023
  ID:H01 NAME:TAN CHIN YONG login
  Delete Account--> ID:S02 NAME:ALI

.Wed Apr  5 17:41:07 2023
  ID:H01 NAME:TAN CHIN YONG login
  Delete Account--> ID:S01 NAME:NAJIB
  Delete Account--> ID:I02 NAME:ALI BABA

.Thu Apr  6 21:46:13 2023
  ID:H01 NAME:TAN CHIN YONG login

.Thu Apr  6 21:47:47 2023
  ID:H01 NAME:TAN CHIN YONG login

.Thu Apr  6 22:00:06 2023
  ID:H01 NAME:TAN CHIN YONG login

.Tue Apr 18 19:01:19 2023
  ID:H01 NAME:TAN CHIN YONG login

.Tue Apr 18 20:30:35 2023
  ID:H01 NAME:TAN CHIN YONG login

.Tue Apr 18 20:36:28 2023
  ID:H01 NAME:TAN CHIN YONG login

.Thu Apr 20 15:03:45 2023
  ID:H01 NAME:TAN CHIN YONG login

.Thu Apr 20 21:26:42 2023
  ID:H01 NAME:TAN CHIN YONG login

.Thu Apr 20 22:00:11 2023
  ID:H01 NAME:TAN CHIN YONG login

.Thu Apr 20 22:19:22 2023
  ID:H01 NAME:TAN CHIN YONG login

.Thu Apr 20 22:27:37 2023
  ID:H01 NAME:TAN CHIN YONG login
  Delete Account--> ID:I01 NAME:O01

.Thu Apr 20 22:37:00 2023
  ID:H01 NAME:TAN CHIN YONG login

.Thu Apr 20 22:40:34 2023
  ID:H01 NAME:TAN CHIN YONG login

.Thu Apr 20 22:41:51 2023
  ID:O01 NAME:RED login

.Thu Apr 20 22:43:39 2023
  ID:F01 NAME:GREEN login

.Thu Apr 20 22:48:37 2023
  ID:H01 NAME:TAN CHIN YONG login

.Thu Apr 20 23:01:59 2023
  ID:H01 NAME:TAN CHIN YONG login

.Thu Apr 20 23:19:22 2023
  ID:H01 NAME:TAN CHIN YONG login

.Thu Apr 20 23:29:44 2023
  ID:H01 NAME:TAN CHIN YONG login

.Thu Apr 20 23:42:00 2023
  ID:H01 NAME:TAN CHIN YONG login

.Sat Apr 22 21:40:39 2023
  ID:H01 NAME:TAN CHIN YONG login
  Delete Account--> ID:I01 NAME:LIM AH BENG

.Sat Apr 22 22:06:05 2023
  ID:H01 NAME:TAN CHIN YONG login

.Sat Apr 22 22:41:15 2023
  ID:H01 NAME:TAN CHIN YONG login

.Sat Apr 22 23:11:40 2023
  ID:H01 NAME:TAN CHIN YONG login

.Sat Apr 22 23:14:12 2023
  ID:H01 NAME:TAN CHIN YONG login

.Sun Apr 23 09:48:52 2023
  ID:H01 NAME:TAN CHIN YONG login

.Sun Apr 23 09:57:50 2023
  ID:H01 NAME:TAN CHIN YONG login

.Sun Apr 23 09:59:55 2023
  ID:H01 NAME:TAN CHIN YONG login

.Sun Apr 23 10:01:58 2023
  ID:H01 NAME:TAN CHIN YONG login

.Sun Apr 23 10:03:36 2023
  ID:H01 NAME:TAN CHIN YONG login

.Sun Apr 23 10:09:16 2023
  ID:H01 NAME:TAN CHIN YONG login

.23-04-2023 12:33
  ID:H01 NAME:TAN CHIN YONG login

.23-04-2023 12:43:35
  ID:H01 NAME:TAN CHIN YONG login


Project100/test.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <conio.h>   //for getch() function
#include<math.h>
#include <time.h>
#include <Windows.h>  //for Sleep() function
#pragma warning(disable:4996)
typedef struct {
	char staffId[7];
	char staffName[31];
	//int staffIc
	int staffPassword[99];
	char staffPosition[26];
}Test;

void maidfwefwef() {
	int r[3];
	printf("%d", r[3]);


	/*Test test[4] = {
		{"abc","def",{57,57,57},"ceo"},
		{"dad","der",{57,57,57},"cio"},
		{"bc","df",{57,57,57},"eo"},
		{"ac","ef",{57,57,57},"co"}
	};
	Test t2[999];
	FILE* fp;
	fp = fopen("t.bin", "rb");
	//fwrite(&test, sizeof(test), 1, fp);
	fread(&t2, sizeof(t2), 1, fp);
	printf("%s", t2[2].staffName);


	fclose(fp);*/
	char c = getch();
	printf("%d", c);
	
}

Project100/main.c
#include "MLM.h"

void main() {
	//displayAllStaff();
	//printf("\n");
	//system("pause");
	//loading();


	char select;
	do {
		sysColor();
		system("cls");
		displayHeader();
		printf("\t\t1.Login as member\n");
		printf("\t\t2.Login as staff\n");
		printf("\t\t3.Notification\n");
		printf("\t\t4.Setting:appearance\n");
		printf("\t\t5.Exit\n\n");
		printf("\t\tEnter > ");

		scanf("%c", &select);
		rewind(stdin);
		switch (select) {
		case '1':
			memberModule();
			break;
		case '2':
			staffModule();
			break;
		case '3':
			displayNotice();
			system("pause");
			break;
		case '4':
			appearance();
			break;
		case '5':
			system("cls");
			char text[50] = "thank for using our system !";
			char text2[100] = "We hope you have a great day and thank you again for choosing our company !";
			strupr(text);
			strupr(text2);
			printf("\n\n\n\n\n\n\n\n\n\n\t\t\t\t\t\t");
			for (int i = 0; i < strlen(text); i++) {
				printf("%c", text[i]);
				Sleep(70);
			}
			printf("\n\t\t\t\t\t\t");
			Sleep(700);
			for (int i = 0; i < strlen(text2); i++) {
				printf("%c", text2[i]);
				if (i == 27) {
					Sleep(500);
					printf("\n\t\t\t\t       ");
				}
				Sleep(70);
			}
			Sleep(1500);
			ending();
			break;
		
		default:
			rewind(stdin);
			printf("\nInvalid\a\n");
			Sleep(1000);
			break;
		}
	} while (select != '5');
}

void appearance() {
	FILE* fp;
	fp = fopen("appearance.bin", "wb");
	int op = '0';
	char key = '0';
	do {
		system("cls");
		printf("\n\t\t\t\t----------------------------------------------\n");
		printf("\t\t\t\t|                Appearance                  |\n");
		printf("\t\t\t\t----------------------------------------------\n");
		printf("\t\t\t\tSelect Then Press Enter...\n");
		printf("\t\t\t\t(Press 'N' to Exit)\n");
		
		switch (key) {
		case '1':
			printf("\n\n\n\t\t\t\t\t=============       -------------");
			printf("\n\t\t\t\t\t|1. Default |         2. Dark  ");
			printf("\n\t\t\t\t\t=============       -------------");
			op = 1;
			break;

		case '2':
		
			printf("\n\n\n\t\t\t\t\t-------------       =============");
			printf("\n\t\t\t\t\t 1. Default         | 2. Dark   |");
			printf("\n\t\t\t\t\t-------------       =============");
			op = 2;
			break;
		default:
			printf("\n\n\n\t\t\t\t\t-------------       -------------");
			printf("\n\t\t\t\t\t 1. Default           2. Dark  ");
			printf("\n\t\t\t\t\t-------------       -------------\n\n");
			op = 0;
			break;
		}
		rewind(stdin);
		key = getch();
		key = toupper(key);
		rewind(stdin);
	} while (key != 'N' && key!=13);
	if (op == 0) {
		fclose(fp);
		return 0;
	}
	else {
		fwrite(&op, sizeof(op), 1, fp);
		loading_animation();
		fclose(fp);
	}

}
void sysColor() {
	int color;
	FILE* fp;
	fp = fopen("appearance.bin", "rb");
	fread(&color, sizeof(color), 1, fp);
	if (color == 1) 
		system("color f0");
	else 
		system("color 0f");
	fclose(fp);
}


void loading() {
	//display loading image
	register int u, l, t;
	t = 0;
	for (int u = 0; u <= 50; u++) {
		system("cls");
		int l = 1;
		printf("\n\n\n\n");
		printf("\t\t                  U _____ u  _        ____   U  ___ u  __  __  U _____ u \n"
			"\t\t      __        __%c| ___%c|/ |%c|    U /%c___ | %c/ %c_ %c/U|' %c/ '|u%c| ___%c|/ \n"
			"\t\t      %c%c%c      /%c/ |  _| %cU | | u  %c| | u     | | | |%c| |%c/| |/ |  _|%c   \n"
			"\t\t      /%c %c /%c / /%c | |___  %c| |/__  | |/__.-,_| |_| | | |  | |  | |___   \n"
			"\t\t     U  %c V  V /  U|_____|  |_____|  %c____|%c_)-%c___/  |_|  |_|  |_____|  \n"
			"\t\t     .-,_%c /%c /_,-.<<   >>  //  %c%c  _// %c%c      %c%c   <<,-,,-.   <<   >>  \n"
			"\t\t      %c_)-'  '-(_/(__) (__)(_%c)(%c_)(__)(__)    (__)   (./  %c.) (__) (__) \n"
			, 92, 34, 34, 34, 92, 34, 92, 92, 92, 34, 92, 34, 92, 34, 34, 92, 92, 92, 34, 92, 92, 92, 92, 92, 92, 92, 92, 92, 92, 92, 92, 92, 92, 92, 92, 92, 92, 34, 34, 92);
		printf("\n\n\n\n\n\n\n\n");
		printf("\t\t\t\t\t\tLoading...%d%c\n\t\t\t    ", t, 37);

		while (l <= u) {
			printf("%c", 22);
			l++;
		}
		printf("%c", 22);
		t += 2;
		Sleep(10);
	}
	printf("\n\n\t\t\t\t\t");

	system("pause");
	system("cls");

}

Project100/sign_up_animation.c
# include <stdio.h>
# include <stdlib.h>
# include <string.h>
# include <ctype.h>
# include <stdbool.h>
# include <conio.h>
# include <Windows.h>
# pragma warning (disable : 4996)
# pragma warning (disable : 6031)

void purple();
void cyan();
void bear();
void congratulations();
void message1();
void message2();

void purple() {
	printf("\033[1;35m");
}

void cyan() {
	printf("\033[0;36m");
}

void bear() {
	system("color 0f");
	cyan();
	printf("			                     ,---.           ,---.\n");
	printf("			                    / /\"`\\.--\"\"\"--./,'\"\\ \\\n");
	printf("			                    \\ \\    _       _    / /\n");
	printf("			                     `./  / __   __ \\  \\,'\n");
	printf("			                      /    /_O)_(_O\\    \\\n");
	printf("			                      |  .-'  ___  `-.  |\n");
	printf("			                   .--|       \\_/       |--.\n");
	printf("			                 ,'    \\   \\   |   /   /    `.\n");
	printf("			                /       `.  `--^--'  ,'       \\\n");
	printf("			             .-\"\"\"\"\"\"-.    `--.___.--'     .-\"\"\"\"\"\"-.\n");
	printf("			.-----------/         \\------------------/         \\--------------.\n");
	printf("			| .---------\\         /----------------- \\         /------------. |\n");
	printf("			| |          `-`--`--'                    `--'--'-'             | |\n");
	printf("			| |                                                             | |\n");
	printf("			| |                                                             | |\n");
	printf("			| |                                                             | |\n");
	printf("			| |");
	congratulations();
	cyan();
	printf("	                | |\n");
	printf("			| |                                                             | |\n");
	printf("			| |                                                             | |\n");
	printf("			| |");
	message1();
	cyan();
	printf("               | |\n");
	printf("			| |                                                             | |\n");
	printf("			| |");                  
	message2();
	cyan();
	printf("		        | |\n");
	printf("			| |                                                             | |\n");
	printf("			| |                                                             | |\n");
	printf("			| |                                                             | |\n");
	printf("			| |_____________________________________________________________| |\n");
	printf("			|_________________________________________________________________|\n");
	printf("			                   )__________|__|__________(\n");
	printf("			                  |            ||            |\n");
	printf("			                  |____________||____________|\n");
	printf("			                    ),-----.(      ),-----.(\n");
	printf("			                  ,'   ==.   \\    /  .==    `.\n");
	printf("			                 /            )  (            \\\n");
	printf("			                 `===========\'    `===========\'\n");
}

void congratulations() {
	purple();
	printf("                     C"); Sleep(70);
	printf("O"); Sleep(70);
	printf("N"); Sleep(70);
	printf("G"); Sleep(70);
	printf("R"); Sleep(70);
	printf("A"); Sleep(70);
	printf("T"); Sleep(70);
	printf("U"); Sleep(70);
	printf("L"); Sleep(70);
	printf("A"); Sleep(70);
	printf("T"); Sleep(70);
	printf("I"); Sleep(70);
	printf("O"); Sleep(70);
	printf("N"); Sleep(70);
	printf("S"); Sleep(70);
	printf("!"); Sleep(70);
}

void message1() {
	purple();
	printf("              Y"); Sleep(70);
	printf("o"); Sleep(70);
	printf("u "); Sleep(70);
	printf("h"); Sleep(70);
	printf("a"); Sleep(70);
	printf("v"); Sleep(70);
	printf("e "); Sleep(70);
	printf("s"); Sleep(70);
	printf("u"); Sleep(70);
	printf("c"); Sleep(70);
	printf("c"); Sleep(70);
	printf("e"); Sleep(70);
	printf("s"); Sleep(70);
	printf("s"); Sleep(70);
	printf("f"); Sleep(70);
	printf("u"); Sleep(70);
	printf("l"); Sleep(70);
	printf("l"); Sleep(70);
	printf("y "); Sleep(70);
	printf("s"); Sleep(70);
	printf("i"); Sleep(70);
	printf("g"); Sleep(70);
	printf("n"); Sleep(70);
	printf("e"); Sleep(70);
	printf("d "); Sleep(70);
	printf("u"); Sleep(70);
	printf("p"); Sleep(70);
	printf("."); Sleep(70);
}

void message2() {
	purple();
	printf("                  W"); Sleep(70);
	printf("e"); Sleep(70);
	printf("l"); Sleep(70);
	printf("c"); Sleep(70);
	printf("o"); Sleep(70);
	printf("m"); Sleep(70);
	printf("e "); Sleep(70);
	printf("t"); Sleep(70);
	printf("o "); Sleep(70);
	printf("o"); Sleep(70);
	printf("u"); Sleep(70);
	printf("r "); Sleep(70);
	printf("c"); Sleep(70);
	printf("o"); Sleep(70);
	printf("m"); Sleep(70);
	printf("p"); Sleep(70);
	printf("a"); Sleep(70);
	printf("n"); Sleep(70);
	printf("y"); Sleep(70);
	printf("!"); Sleep(70);
}

Project100/MLM.h
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>
#include <conio.h>
#include<math.h>
# include <stdbool.h>
#include <windows.h>
#include <ctype.h>
#pragma warning (disable:4996)
#pragma warning (disable:6031)
void appearance();
void sysColor();

//               Structure
//1.stock module's structure
typedef struct {
	char item_code[10];
	char item_description[100];
	float item_price;
	int item_quantity, minimum_level, reorder_quantity;
}StockItem;

//2.staff module's structure
typedef struct {
	int day, month, year;
}Date;
typedef struct {
	int hour, minute, second;
}Htime;
typedef struct {
	char houseNo[26];
	char streetNo[26];
	char postCode[6];
	char city[26];
	char state[26];
} Address;
typedef struct {
	char staffId[4];
	char staffName[31];
	char staffGender;
	char staffIc[20];
	int staffPassword[99];
	char staffPosition[26];
	Address staffAddress;
	char joinDate[31];
	char lastLogin[31];
	Date birth;
}Staffaccount;
typedef struct {
	Date date;
	Htime htime;
	char staffId[4];
	char staffName[31];
	char event[99];
}History;

//3.member module's structure
typedef struct {
	long int sec;
} BlockingTime;
struct Member {
	char member_ID[6];
	char username[100];
	char gender[50];
	char IC[15];
	char email[100];
	char upline[100];
	char password[51];
	char password_confirm[51];
	int key;
	char encrypted_password[51];
	int question;
	char answer[100];
	char account_available[10];
	Address add;
	BlockingTime time;
};




//                 Functions
//-----------------------------------------------------------------------------------------------------------------------------
//1.stock module's function
StockItem stockItem[1000];
int numOfItem;
char cur_time[30];
void stockModule();
void add_item_to_stock_rand_generate();
void upline_purchase();
void display_asc_price();
void display_desc_price();
void display_asc_qty();
void display_desc_qty();
void import_file_to_add_stock();
void read_stock_from_file();
void delete_item_by_item_code();
int search_item_code(char item_code[]);
void search_item_by_price();
void get_lastest_time();
void write_stock_to_file();
void update_stock_item();
void add_item_to_stock();
void search_between_quantity();
void search_stock_item_code();
void display_stock();
void curTime(char timevalue[30]);
void minimum_stock_alert();
void loading_animation();
//-----------------------------------------------------------------------------------------------------------------------------
//2.staff module's function
void displayStaffHeader();
Staffaccount stafflogin(int c);
void banana();
void alien();
void dog();
void cat();
int  verify_robot();
void displayAllStaff();
void registerNewStaff(int staffCount, Staffaccount currentAcc);
void loading();
void displayHeader();
void getTime(char timevalue[30]);
int confirmPassword(int passwords[99], int count);
int countstaff(int countstaff, Staffaccount staff);
int check_Position_Existing(char position[26]);
void deletestaff(int c, Staffaccount currentAcc);
void modifyStaffAcc(int c, Staffaccount currentAcc);
void staffModule();
void display_currentAcc_info(Staffaccount userInfo);
void ending();
void displayHistory();
void countNotice();
void displayNotice();
void manageNotice(Staffaccount currentAcc);
void addNotice(Staffaccount currentAcc);
void deleteNotice(Staffaccount currentAcc);
void getDateTime(int* day, int* month, int* year, int* hour, int* min, int* second);
void resetPassword(Staffaccount currentAcc);


//3.member module
//-----------------------------------------------------------------------------------------------------------------------------
int view_memberlist();
int delete_member(struct Member read[1000], int count);
int view_bannedlist();
int restore_member(struct Member read[1000], int count);
void red();
void searching_animation();
void deleteMember();

//4.sale module
//-----------------------------------------------------------------------------------------------------------------------------
void memberModule();
#define _WIN32_WINNT 0x0500
#define COMMISSION 0.05



struct ShoppingCart {
	int cartID, custID;
	char sCartID[50];
};

typedef struct {
	char prodID[6];
	int qty;
	char prodName[50];
	double price;
	char status[50];
} Product;

typedef struct {
	char prodID[6];
	char prodName[50];
} BuyAgain;

typedef struct {
	WORD day, month, year, hour, min, sec;
	char prodID[6];
	int qty;
	char prodName[50];
	double price;
	char status[50];
} ViewHistory;

// Member Module
int signup();
char* login();
bool username_validation(struct Member info);
int gender_validation(struct Member info);
int icnum_validation(struct Member info);
int email_validation(struct Member info);
int password_validation(struct Member info);
int confirm_password(struct Member info);
char* password_encryption(struct Member info);
char* password_decryption(struct Member input, char encrypted_password[51]);
int security_question();
void output(struct Member info);
int view_profile(char username[100]);
char* edit_profile(char name[100]);
int forget_password();
char* password_reset();
int view_product(char username[100]);
int search_product(StockItem prod[1000], StockItem prodSearched[1000], int count, int m, char username[100]);
int purchase_product(StockItem prod[1000], StockItem prodSearched[1000], int count, int m, char username[100]);
int view_stock(char username[100]);
void output2(struct Member info);
void red();
void searching_animation();
void purple();
void cyan();
void bear();
void congratulations();
void message1();
void message2();

// Sales Module
int salesModule(char memberID[]);
char module(char memberID[]);
char displayShoppingCart(char memberID[]);
void shoppingCart(char prodID[], char prodName[], int qty, double price, char memberID[]);
int buyAgain(char memberID[]);
void rShopCart(char memberID[]);
int modify(char memberID[]);
int delete(char memberID[]);
int checkoutAll(char memberID[]);
int checkoutSelect(char memberID[]);
int selectBuy(char* checkout, char memberID[]);
int buyNow(char prodID[], char prodName[], int qty, double price, char memberID[]);
double totalPrice(FILE* order);
void placeOrder(char shopCart[], char buyAgain[], char temp[], char memberID[]);
void reduceStock(char shopCart[], char memberID[]);
void commission(char shopCart[], char memberID[]);
void purchaseHistory(char buyCart[], char memberID[]);
char viewHistory(char memberID[]);
int search(ViewHistory search[], FILE* view);
